src/ast/pass.h

```diff
@@ -0,0 +1,34 @@
+// Copyright Microsoft and Project Verona Contributors.
+// SPDX-License-Identifier: MIT
+#pragma once
+
+#include "ast.h"
+#include "err.h"
+
+#include <vector>
+
+namespace pass
+{
+  /// A Pass object has a name and an operation, that is executed on AST nodes.
+  class Pass
```

Comment:
Personally, I would just use a `std::tuple` here. Of course it is fine as it is!

Reply:
Personal preference. I think it's marginally easier to read the code (`pass.name` rather than `get<0>(pass)` and `pass(args)` rather than `get<1>(pass)(args)`).